src/clipping/weileratherton.jl

```diff
@@ -0,0 +1,203 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    WeilerAtherton()
+
+The Weiler-Atherton algorithm for clipping polygons.
+
+## References
+
+* Weiler, Kevin and Atherton, Peter. 1977. [Hidden surface removal using polygon area sorting]
+  (https://dl.acm.org/doi/10.1145/563858.563896)
+"""
+struct WeilerAtherton <: ClippingMethod end
+
+function clip(poly::Polygon{Dim,T}, other::Geometry{Dim,T}, method::WeilerAtherton) where {Dim, T}
+  if hasholes(poly)
+    outers = clip(rings(poly)[1], boundary(other), method)
+    inners = vcat([clip(r, boundary(other), method) for r in rings(poly)[2:end]]...)
```

Comment:
Try to avoid branching here. You can simply apply the method to all rings using a single vector.

Then later you can decide what to do based on the length of the vector of rings.